mitodl/micromasters
# Removed routing from initial profile flow

#### What are the relevant tickets?

closes #627 

#### What's this PR do?

This changes the way we progress through the initial profile form sequence. Currently, there's a separate url for each component (`/profile/personal -> PersonalTab`, etc), with the `/profile` route rendering the `ProfilePage` component and the tab component routes nested inside. This changes things so that:

- there's a redux action added to the ui reducer to keep track of what step the user is on (`profileStep`)
- `ProfilePage` pays attention to this bit of state and selectively renders the right tab component
- `ProfilePage` also passes in `nextStep` and `prevStep` callbacks (which either update the store or cause a react-router route transition) which the tab components can pass to `ProfileProgressControls`
- generating props for child components on `ProfileFormContainer` is pulled out into a separate method, which means we can use it either in the case of setting props on cloned children or on rendering a form component without sub-routes
- `SettingsPage` also adopts some of this pattern (using the `profileProps` method from `ProfileFormContainer`) and is, I think, a bit simpler

I think that's it...a lot of changes were necessary to our existing tests for `ProfilePage` and `App` as well.

#### Where should the reviewer start?

Review the changes to `ProfilePage` and `ProfileFormContainer` first, I think, and then look over the changes to existing tests and so on.

#### How should this be manually tested?

There should be no regressions in functionality on the profile, so:

- should be able to edit stuff
- backward / forwards buttons should work
- button labels should change on the privacy tab
- there shouldn't be any appearance changes
- additionally the settings page shouldn't have any regressions

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
